Scripts/Editor/Utils/CodeGenerationUtility.cs

```diff
@@ -182,27 +183,55 @@ public static void AppendFooter(StreamWriter writer, ref int indentation, string
             }
         }
 
+        public static void ValidateIfCanBePartial(ScriptableObjectCollection collection)
+        {
+            SerializedObject collectionSerializedObject = new SerializedObject(collection);
+
+            bool canBePartial = CheckIfCanBePartial(collection);
+            if (collectionSerializedObject.FindProperty("generateAsPartialClass").boolValue && !canBePartial)
```

Comment:
Nitpick: you're finding the property twice instead of doing it once and caching it. Duplication!